services/client/services_test.go

```diff
@@ -136,6 +136,16 @@ func TestAdvertiseErrors(t *testing.T) {
 	err = sd.Advertise(ad)
 	require.NoError(t, err)
 
+	hbGen.Lock()
+	hb := hbGen.hbs[serviceKey(sid)]
+	hbGen.Unlock()
+
+	// hb store should show advertisement (almost) immediately
+	time.Sleep(time.Millisecond)
```

Comment:
the test case could be: construct the sd client with a super long HeartbeatInterval, like an hour, and verify the heartbeat still got created pretty soon